packages/components-js/tests/e2e/specs/checkbox.e2e.ts

```diff
@@ -0,0 +1,623 @@
+
+  await setProperty(host, 'state', 'none');
+  await setProperty(host, 'message', '');
+  await waitForStencilLifecycle(page);
+
+  await expect(getMessage(page), 'when state = none').toHaveCount(0);
+});
+
+test('should toggle checkbox when input is clicked', async ({ page }) => {
+  await initCheckbox(page);
+  const host = getHost(page);
+  const input = getInput(page);
+
+  expect(await getBackgroundImage(input)).toBe('none');
+
+  await input.click();
+
+  const checkedImage = await getBackgroundImage(input);
+  expect(checkedImage).toContain(backgroundURL);
+
+  await input.click();
+  expect(await getBackgroundImage(input)).toBe('none');
+
+  // ensure that checked and indeterminate use different images
+  await setProperty(host, 'indeterminate', true);
+  expect(checkedImage).not.toBe(await getBackgroundImage(input));
+});
+
+test('should not toggle checkbox on click in loading state', async ({ page }) => {
+  await initCheckbox(page, { loading: true });
+  const host = getHost(page);
+  const input = getInput(page);
+  await addEventListener(host, 'update');
+
+  await expect(host).toHaveJSProperty('checked', false);
+  await input.click({ force: true });
+  await expect(host).toHaveJSProperty('checked', false);
+
+  await performBoundaryClicks(host, page);
+
+  expect((await getEventSummary(host, 'update')).counter).toBe(0);
+
+  await setProperty(host, 'loading', false);
+  await waitForStencilLifecycle(page);
+
+  await input.click();
+  await expect(host).toHaveJSProperty('checked', true);
+
+  expect((await getEventSummary(host, 'update')).counter).toBe(1);
+});
+
+test('should not toggle checkbox on click in disabled state', async ({ page }) => {
+  await initCheckbox(page, { disabled: true });
+  const host = getHost(page);
+  const input = getInput(page);
+  await addEventListener(host, 'update');
+
+  await expect(host).toHaveJSProperty('checked', false);
+  await input.click({ force: true });
+  await expect(host).toHaveJSProperty('checked', false);
+
+  await performBoundaryClicks(host, page);
+
+  expect((await getEventSummary(host, 'update')).counter).toBe(0);
+
+  await setProperty(host, 'disabled', false);
+  await waitForStencilLifecycle(page);
+
+  await input.click();
+  await expect(host).toHaveJSProperty('checked', true);
+
+  expect((await getEventSummary(host, 'update')).counter).toBe(1);
+});
+
+test.describe('focus', () => {
+  test('should receive focus', async ({ page }) => {
+    await initCheckbox(page);
+    const input = getInput(page);
+
+    await input.click();
+    expect(await getActiveElementTagName(page)).toBe('P-CHECKBOX');
+  });
+
+  test('should receive focus in loading state', async ({ page }) => {
+    await initCheckbox(page, { loading: true });
+    const input = getInput(page);
+
+    await input.focus();
+    expect(await getActiveElementTagName(page)).toBe('P-CHECKBOX');
+  });
+
+  test('should receive focus when focus is set programmatically', async ({ page }) => {
```

Comment:
Since we just talked about this topic. Maybe would be good to also test the style change here as well.